lib/src/util/boolean_expression_utilities.dart

```diff
@@ -10,6 +10,15 @@ class BooleanExpressionUtilities {
   static HashSet<TokenType> BOOLEAN_OPERATIONS =
       HashSet.from(const [TokenType.AMPERSAND_AMPERSAND, TokenType.BAR_BAR]);
 
+  static HashSet<TokenType> ALL_COMPARISONS_OPERATIONS = HashSet.from(const [
```

Comment:
I'd drop the prefix "ALL_" from the name. I don't think it clarifies the meaning of the name, and the form without it is more consistent with the names of the other constants in this class.

Speaking of which, I'd argue that it would be better to be inconsistent and use a set literal here, which would allow this field to be marked as a `const`. I suspect that the other fields here were created before set literals existed. They should be converted at some point, but I'm not asking you to do that as part of this PR.

Reply:
obsolete, because i am now following bwilkerson to only check for equality == 0/1 operator for this lint.